grammars/javascript.cson

```diff
@@ -17,8 +17,162 @@
+  }
+  {
+    # foo: function() { … }
+    'begin': '''(?x)
+      (\\b([a-zA-Z_?.$][\\w?.$]*))
+      \\s*(:)\\s*
+      \\b(?:(async)(?:\\s+))?
+      (function\\*?)
+      \\s*(\\*?)\\s*
+      ([a-zA-Z_?$][\\w?$]*)?
+      \\s*(\\()
+    '''
+    'beginCaptures':
+      '1':
+        'name': 'meta.object-key.js'
+      '2':
+        'name': 'entity.name.function.js'
+      '3':
+        'name': 'keyword.operator.assignment.js'
+      '4':
+        'name': 'storage.modifier.js'
+      '5':
+        'name': 'storage.type.function.js'
+      '6':
+        'name': 'storage.type.function.js'
+      '7':
+        'name': 'entity.name.function.js'
+      '8':
+        'name': 'punctuation.definition.parameters.begin.js'
+    'end': '(\\))'
+    'endCaptures':
+      '1':
+        'name': 'punctuation.definition.parameters.end.js'
+    'name': 'meta.function.json.js'
+    'patterns': [
+      {
+        'include': '#function-params'
+      }
+    ]
+  }
+  {
+    # "foo": function() { … }
+    'begin': '''(?x)
+      ((?:((\')([^\']*?)(\'))|((")([^"]*?)("))))
+      \\s*(:)\\s*
+      \\b(?:(async)(?:\\s+))?
+      (function\\*?)
+      \\s*(\\*?)\\s*
+      ([a-zA-Z_?$][\\w?$]*)?
+      \\s*(\\()
+    '''
+    'beginCaptures':
+      '1':
+        'name': 'meta.object-key.js'
+      '2':
+        'name': 'string.quoted.single.js'
+      '3':
+        'name': 'punctuation.definition.string.begin.js'
+      '5':
+        'name': 'punctuation.definition.string.end.js'
+      '6':
+        'name': 'string.quoted.double.js'
+      '7':
+        'name': 'punctuation.definition.string.begin.js'
+      '9':
+        'name': 'punctuation.definition.string.end.js'
+      '10':
+        'name': 'keyword.operator.assignment.js'
+      '11':
+        'name': 'storage.modifier.js'
+      '12':
+        'name': 'storage.type.function.js'
+      '13':
+        'name': 'storage.type.function.js'
+      '14':
+        'name': 'entity.name.function.js'
+      '15':
+        'name': 'punctuation.definition.parameters.begin.js'
+    'end': '(\\))'
+    'endCaptures':
+      '1':
+        'name': 'punctuation.definition.parameters.end.js'
+    'name': 'meta.function.json.js'
+    'patterns': [
+      {
+        'include': '#function-params'
+      }
+    ]
+  }
+  {
+    'begin': '\\?'
+    'beginCaptures':
+      '0':
+        'name': 'keyword.operator.ternary.js'
+    'end': ':'
+    'endCaptures':
+      '0':
+        'name': 'keyword.operator.ternary.js'
+    'patterns': [
+      {
+        'match': '(\\w+)(?=\\s*:)'
+        'captures':
+          '1':
+            'patterns': [
+              'include': '$self'
+            ]
+      }
+      {
+        'match': '(?:"[^"]*")|(?:\'[^\']*\')(?=\\s*:\\s*\\.*)'
+        'captures':
+          '0':
+            'patterns': [
+              'include': '#strings'
+            ]
+      }
+      {
+        'include': '$self'
+      }
+    ]
+  }
+  {
+    'match': '(?:^\\s*)([a-zA-Z_$][\\w$]*)\\s*(:)(?=\\n)'
```

Comment:
Will this capture succeed if there's a comment after the label, eg `label: /* comment! */`?